ui/src/app/api/auth/signup/page.tsx

```diff
@@ -0,0 +1,30 @@
+"use client";
+import AutoForm, { AutoFormSubmit } from "@/components/ui/auto-form";
+import axios from "axios";
+import * as z from "zod";
+import { FormSchema } from "./zod.signup.schema";
+import { useRouter } from 'next/navigation'
+
+export default function InputForm() {
+  async function handleSubmit(data: z.infer<typeof FormSchema>) {
+    try {
+    const response = await axios.post('http://localhost:3333/auth/signup', data)
+    console.log(response.statusText)
+    if (response.statusText === 'Created') {
+      const router = useRouter()
```

Comment:
Using hooks like this not in top level of component may lead to errors: https://legacy.reactjs.org/docs/hooks-rules.html the useRouter needs to be called in InputForm directly